pandas/core/indexing.py

```diff
@@ -993,7 +993,9 @@ def _validate_key(self, key, axis: int):
         # slice of labels (where start-end in labels)
         # slice of integers (only if in the labels)
         # boolean not in slice and with boolean index
-        if isinstance(key, bool) and not is_bool_dtype(self.obj.index):
+        if isinstance(key, bool) and not (
+            is_bool_dtype(self.obj.index) or self.obj.index.dtype.name == "boolean"
```

Comment:
That might need a fix in `ìs_bool_dtype` instead to recognize Index[boolean] ? 

(it currently checks for Index is it is object dtype, and then checks the inferred dtype, but so that needs to be updated now also non-object dtype Index can be boolean)

Reply:
Can you check this comment? We currently have an explicit check for Index objects that only considers object dtype:

https://github.com/pandas-dev/pandas/blob/5ee8cf1daed481bdfd28399468a1233736332db3/pandas/core/dtypes/common.py#L1316-L1322

You can probably for non-object dtypes just check ``arr_or_dtype.inferred_type == "boolean"`` (or explicitly check the name of the dtype, or check the `_is_boolean` attribute of the dtype for EAs)